pkg/vault/test/foundry/fork/YieldBearingPoolSwapBase.t.sol

```diff
@@ -0,0 +1,1097 @@
+            ybToken2.convertToShares(_token2BufferInitAmount),
+            "Wrong ybToken2 buffer balance for wrapped token"
+        );
+    }
+
+    function testToken1InToken2OutWithinBufferExactIn__Fork__Fuzz(uint256 amountIn) public {
+        // Test from 10% to 50% of tokenIn's buffer to avoid exceeding the limit of tokenOut's buffer.
+        amountIn = bound(amountIn, (_token1BufferInitAmount) / 10, _token1BufferInitAmount / 2);
+        IBatchRouter.SwapPathExactAmountIn[] memory paths = _buildExactInPaths(_token1Fork, amountIn, 0);
+
+        int256 expectedBufferDeltaTokenIn = int256(amountIn);
+
+        uint256 wrappedBufferDeltaTokenIn = ybToken1.convertToShares(amountIn);
+        uint256 wrappedBufferDeltaTokenInScaled18 = wrappedBufferDeltaTokenIn.divDown(_ybToken1Factor);
+        // PoolMock is linear, so wrappedAmountInScaled18 = wrappedAmountOutScaled18
+        uint256 wrappedBufferDeltaTokenOutRaw = wrappedBufferDeltaTokenInScaled18.mulDown(_ybToken2Factor);
+        int256 expectedBufferDeltaTokenOut = -int256(ybToken2.convertToAssets(wrappedBufferDeltaTokenOutRaw));
+
+        _testExactIn(paths, true, expectedBufferDeltaTokenIn, expectedBufferDeltaTokenOut);
+    }
+
+    function testToken1InToken2OutWithinBufferExactOut__Fork__Fuzz(uint256 amountOut) public {
+        // Test from 10% to 50% of tokenOut's buffer to avoid exceeding the limit of tokenIn's buffer.
+        amountOut = bound(amountOut, (_token2BufferInitAmount) / 10, _token2BufferInitAmount / 2);
+        IBatchRouter.SwapPathExactAmountOut[] memory paths = _buildExactOutPaths(
+            _token1Fork,
+            (2 * amountOut * _token1Factor) / _token2Factor,
+            amountOut
+        );
+
+        uint256 wrappedBufferDeltaTokenOut = ybToken2.convertToShares(amountOut);
+        uint256 wrappedBufferDeltaTokenOutScaled18 = wrappedBufferDeltaTokenOut.divDown(_ybToken2Factor);
+        // PoolMock is linear, so wrappedAmountInScaled18 = wrappedAmountOutScaled18
+        uint256 wrappedBufferDeltaTokenInRaw = wrappedBufferDeltaTokenOutScaled18.mulDown(_ybToken1Factor);
+        int256 expectedBufferDeltaTokenIn = int256(ybToken1.convertToAssets(wrappedBufferDeltaTokenInRaw));
+
+        int256 expectedBufferDeltaTokenOut = -int256(amountOut);
+
+        _testExactOut(paths, true, expectedBufferDeltaTokenIn, expectedBufferDeltaTokenOut);
+    }
+
+    function testToken1InToken2OutOutOfBufferExactIn__Fork__Fuzz(uint256 amountIn) public {
+        // Test from 2x to 4x of tokenIn's buffer to make sure it's out of both buffer ranges but yield-bearing pool
+        // has enough tokens to trade (5x buffer init amount).
+        amountIn = bound(amountIn, 2 * _token1BufferInitAmount, 4 * _token1BufferInitAmount);
+        IBatchRouter.SwapPathExactAmountIn[] memory paths = _buildExactInPaths(_token1Fork, amountIn, 0);
+
+        // Since operation is out of buffer range, buffer balances should not change (wrap/unwrap tokens directly).
```

Comment:
This happens because we start balanced, and we just wrap / unwrap right?

Reply:
Yes, I adjusted the comment.